numba/targets/optional.py

```diff
@@ -89,6 +85,7 @@ def optional_to_optional(context, builder, fromty, toty, val):
 
 
 @lower_cast(types.Any, types.Optional)
+@lower_cast(types.Boolean, types.Optional)
```

Comment:
The fact that we need to specialize for `(boolean, optional)` is weird. What is the ambiguity here?